repos/system_upgrade/common/actors/cloud/checkhybridimage/tests/test_checkhybridimage.py

```diff
@@ -27,56 +26,70 @@
 EFI_PATH = '/boot/efi/EFI/redhat/grubenv'
 
 
-def test_hybrid_image(monkeypatch, tmpdir):
-    grubenv_efi = tmpdir.join('grubenv_efi')
-    grubenv_efi.write('grubenv')
+def raise_call_error(args=None):
+    raise CalledProcessError(
+        message='A Leapp Command Error occurred.',
+        command=args,
+        result={'signal': None, 'exit_code': 1, 'pid': 0, 'stdout': 'fake', 'stderr': 'fake'}
+    )
 
-    grubenv_boot = tmpdir.join('grubenv_boot')
-    grubenv_boot.mksymlinkto('grubenv_efi')
 
-    monkeypatch.setattr(checkhybridimage, 'BIOS_PATH', grubenv_boot.strpath)
-    monkeypatch.setattr(checkhybridimage, 'EFI_PATH', grubenv_efi.strpath)
-    monkeypatch.setattr(reporting, 'create_report', create_report_mocked())
-    monkeypatch.setattr(
-        api, 'current_actor', CurrentActorMocked(arch='x86_64', msgs=[BIOS_FIRMWARE, INSTALLED_AGENT])
-    )
-    monkeypatch.setattr(api, "produce", produce_mocked())
+class run_mocked(object):
+    def __init__(self, hypervisor='', raise_err=False):
+        self.hypervisor = hypervisor
+        self.called = 0
+        self.args = []
+        self.raise_err = raise_err
+
+    def __call__(self, *args):  # pylint: disable=inconsistent-return-statements
+        self.called += 1
+        self.args.append(args)
+
+        if self.raise_err:
+            raise_call_error(args)
+
+        if args[0] == ['systemd-detect-virt']:
+            return {'stdout': self.hypervisor}
```

Comment:
We should be explicit about our expectations and raise an error if some unexpected command is executed, imho.
```suggestion
        assert args[0] == ['systemd-detect-virt']
        return {'stdout': self.hypervisor}
```

Reply:
makes sense. fixed!